redis_json/src/ivalue_manager.rs

```diff
@@ -140,25 +139,24 @@ fn update<F: FnMut(&mut IValue) -> Result<Option<()>, Error>>(
             }
             ValueType::Array => {
                 let arr = target_once.as_array_mut().unwrap();
-                if let Ok(x) = token.parse::<usize>() {
-                    if is_last {
-                        if x < arr.len() {
-                            let v = &mut arr.as_mut_slice()[x];
-                            match (func)(v) {
-                                Ok(res) => {
-                                    if res.is_none() {
-                                        arr.remove(x);
-                                    }
+                let x = token.parse::<usize>().expect(
+                    "Array index should have been parsed successfully before reaching here",
+                );
```

Comment:
Can we shorten the message here as in another example and call the variable `index` instead, please?